### PR TITLE
chef server only deploy

### PR DESCRIPTION
**Pull request by jaym**
# How it works
```
# cat chef_server.toml 
[global.v1.external.automate]
enable = true
endpoint = "http://localhost:15555"
[global.v1.external.automate.auth]
token = "thisisatoken"

[deployment.v1.svc]
products = ["chef-server"]

[erchef.v1.sys.data_collector]
enabled = true


# chef-automate config patch chef_server.toml
...

# chef-automate status

Status from deployment with channel [dev] and type [local]

Service Name           Process State  Health Check  Uptime (s) PID  
deployment-service     running        ok            1026       26334
backup-gateway         running        ok            1474       5602 
automate-postgresql    running        ok            1473       5768 
automate-pg-gateway    running        ok            1473       5703 
automate-elasticsearch running        ok            1473       5655 
automate-es-gateway    running        ok            1473       5727 
pg-sidecar-service     running        ok            1473       5797 
es-sidecar-service     running        ok            1465       7000 
automate-cs-bookshelf  running        ok            1023       26522
automate-cs-oc-bifrost running        ok            1022       26557
automate-cs-oc-erchef  running        ok            1023       28747
automate-cs-nginx      running        ok            1022       26590
automate-load-balancer running        ok            1461       7369 

```

# Tasks
- [x] Can deploy only chef server
- [x] [Backups](https://github.com/chef/automate/projects/1#card-22652642)
- [x] [Configure external data collector](https://github.com/chef/automate/projects/1#card-22652710)
- [x] Cleanup
- [x] Tests

### CODE REVIEW

**Comment by jaym** [comment]
Should components be called something else (our code uses collections) and components means something else inside our repo

**Comment by jaym** [comment]
people like `products` better than `collections`

File: components/automate-cs-oc-erchef/habitat/hooks/run [hunk 1]
```diff
@@ -55,6 +55,13 @@ export GEM_PATH={{pkgPathFor "core/bundler"}}
 # non root svc user and we need write access to the file
 cp {{pkg.svc_config_path}}/dark_launch_features.json {{pkg.svc_data_path}}
 
+{{#if cfg.external_automate.enable}}
+# TODO(jaym): Our config should know how to deal with the token. We shouldn't be showing
```
Comment (ryancragun) [inline components/automate-cs-oc-erchef/habitat/hooks/run]: should we resolve this todo before merging?

Comment (jaym) [reply components/automate-cs-oc-erchef/habitat/hooks/run]: This is a fairly large chunk of work. We'd need to teach the config what's a secret and then figure out how to store it in secrets service

File: api/config/cs_nginx/config_request.proto [hunk 2]
```diff
@@ -76,6 +77,15 @@ message ConfigRequest {
 				// Auto-calculated
 				google.protobuf.StringValue content_md5 = 3;
 			}
+
+			message ExternalAutomate {
+				google.protobuf.BoolValue enable = 1;
+				google.protobuf.BoolValue ssl_upstream = 2;
```
[the remark this answers is not in the document]
Comment (jaym) [reply api/config/cs_nginx/config_request.proto]: i think for consistency with the rest of the global config, we should keep it as ssl:
https://github.com/chef/automate/blob/master/api/config/shared/global.proto#L57
https://github.com/chef/automate/blob/master/api/config/shared/global.proto#L88

File: .expeditor/verify_private.pipeline.yml [hunk 3]
```diff
@@ -518,6 +518,15 @@ steps:
         linux:
           privileged: true
 
+  - label: "[integration] chef server only"
+    command:
+      - integration/run_test integration/tests/chef_server_only.sh
+    timeout_in_minutes: 30 # longer timeout for chef-server
```
[the remark this answers is not in the document]
Comment (jaym) [reply .expeditor/verify_private.pipeline.yml]: copy-pasta

File: api/config/shared/global.proto [hunk 4]
```diff
@@ -100,8 +100,26 @@ message External {
...
+			google.protobuf.StringValue token = 2;
+		};
+
+		message SSL {
+			google.protobuf.StringValue root_cert = 1;
+			google.protobuf.StringValue server_name = 2;
+		};
+
+		google.protobuf.BoolValue enable = 1;
+		google.protobuf.StringValue node = 2;
```
Comment (ryancragun) [inline api/config/shared/global.proto]: I'm not sure I like `node` here as we treat it as an endpoint url. Perhaps something like `endpoint` or `url` would be better?

File: components/automate-cs-nginx/habitat/config/nginx.conf [hunk 5]
```diff
@@ -83,12 +83,21 @@ http {
     keepalive 8;
   }
 
+{{#if cfg.external_automate.enable}}
+  upstream external-automate {
+    server {{cfg.external_automate.endpoint}};
+    keepalive 8;
```
Comment (ryancragun) [inline components/automate-cs-nginx/habitat/config/nginx.conf]: I'm all for this for now but I'm curious to see how keepalives against an external automate will work in practice.

File: integration/tests/chef_server_only.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    cat << EOF >> "$test_config_path"
+[erchef.v1.sys.data_collector]
+enabled = false
+EOF
+}
+
+do_deploy() {
+    chef-automate deploy config.toml \
+        --collection chef-server \
```
Comment (ryancragun) [inline integration/tests/chef_server_only.sh]: Think this should be `--product`

File: api/config/shared/global.proto (hunk 4 above)
Comment (jaym) [reply api/config/shared/global.proto]: We called things node for elasticsearch and postgres

**Pull request merged**